lib/progressrus.rb

```diff
@@ -2,29 +2,60 @@
 require 'securerandom'
 require 'redis'
 require 'time'
-require_relative "progressrus/store"
 require_relative "progressrus/store/base"
 require_relative "progressrus/store/redis"
 require_relative "progressrus/store/progressbar"
 require_relative "progressrus/core_ext/enumerable"
 
 class Progressrus
+  class InvalidStoreError < StandardError
+    def initialize
+      message = <<~MSG
+        The store needs to implement `persist`, `scope`, `find` and `flush`
+        We have a base class that your store can inherit from:
+          Progressrus::Store::Base
+      MSG
+      super(message)
+    end
+  end
+
   class << self
+    def clear_stores
+      @stores = {}
+    end
+
     def stores
-      @@stores ||= Store.new(Store::Redis.new(::Redis.new(host: ENV["PROGRESSRUS_REDIS_HOST"] || "localhost")))
+      @stores ||= {
+        redis: Store::Redis.new(::Redis.new(host: ENV["PROGRESSRUS_REDIS_HOST"] || "localhost"))
+      }
     end
 
-    def scope(scope, store: :first)
-      stores.find_by_name(store).scope(scope)
+    def add_store(name, store)
+      validate_store!(store)
+      new_stores = stores.merge(name => store)
```

Comment:
Why `merge` if you could do `stores[name] = store`?
Mutating is not always good, but in this case I don't see any reason not to do it.
`Progressrus.stores` is ok to be mutated directly.